File: backbone-go/examples/clean-api-go/domain/specifications/product_specs.go
// Package specifications contains domain-specific product specifications
package specifications

import (
	"github.com/freakjazz/backbone-go/domain/specifications"
)

// ProductIsActive checks if a product is active
func ProductIsActive() specifications.Specification {
	return specifications.NewEqualSpecification("active", true)
}

// ProductInCategory checks if a product is in a specific category
func ProductInCategory(category string) specifications.Specification {
	return specifications.NewEqualSpecification("category", category)
}

// ProductPriceRange checks if product price is in range
func ProductPriceRange(min, max float64) specifications.Specification {
	return specifications.NewBetweenSpecification("price", min, max)
}

// ProductInStock checks if product is in stock
func ProductInStock() specifications.Specification {
	return specifications.NewGreaterThanSpecification("stock", 0)
}

// ProductByNamePattern searches products by name pattern
func ProductByNamePattern(pattern string) specifications.Specification {
	return specifications.NewLikeSpecification("name", pattern)
}

File: backbone-python/examples/clean_api_python/domain/specifications/product_specs.py
"""Product specifications - Domain layer"""
import sys
import os

# Add backbone to path
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../../../..')))

from backbone.domain.specifications import FilterSpecification


def product_is_active() -> FilterSpecification:
    """Checks if a product is active"""
    return FilterSpecification("active", "=", True)


def product_in_category(category: str) -> FilterSpecification:
    """Checks if a product is in a specific category"""
    return FilterSpecification("category", "=", category)


def product_price_range(min_price: float, max_price: float) -> FilterSpecification:
    """Checks if product price is in range"""
    return FilterSpecification("price", ">=", min_price) & FilterSpecification("price", "<=", max_price)


def product_in_stock() -> FilterSpecification:
    """Checks if product is in stock"""
    return FilterSpecification("stock", ">", 0)


def product_by_name_pattern(pattern: str) -> FilterSpecification:
    """Searches products by name pattern"""
    return FilterSpecification("name", "LIKE", f"%{pattern}%")
